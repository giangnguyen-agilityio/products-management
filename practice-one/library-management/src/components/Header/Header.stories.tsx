import { type StoryFn, type Meta } from '@storybook/react'
import Header from './index'

export default {
  title: 'Example/Header',
  component: Header
} as Meta<typeof Header>

const Template: StoryFn<typeof Header> = (args) => <Header {...args} />
export const Default = Template.bind({})
Default.args = {
  isLogin: false
}
