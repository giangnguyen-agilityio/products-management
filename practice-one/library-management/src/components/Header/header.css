.site-header-content {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.site-navigation {
  display: grid;
  grid-template-columns: 1fr 250px;
  gap: 100px;
}

.site-actions {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 30px;
  height: 100%;
  align-items: center;
}

.sign-up {
  background-color: unset !important;
  color: var(--primary-color) !important;
}

.sign-up:hover {
  background-color: var(--bg-btn-color) !important;
}

.sign-up,
.sign-in {
  padding: 15px !important;
  border-radius: 25px !important;
  transition: all 0.3s ease-in-out;
}
