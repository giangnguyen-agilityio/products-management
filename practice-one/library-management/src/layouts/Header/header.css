.site-header-content {
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 2px solid var(--primary-color);
}

.site-navigation {
  display: grid;
  align-items: center;
  grid-template-columns: 1fr 250px;
  gap: 100px;
}

.site-actions {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 30px;
  height: 100%;
  align-items: center;
}

.sign-up {
  background-color: unset;
  color: var(--primary-color);
}

.sign-up:hover {
  background-color: var(--bg-btn-color);
}

.sign-up,
.sign-in {
  padding: 15px;
  border-radius: 25px;
  transition: all 0.3s ease-in-out;
}

/* Media Query for Laptop Devices */
@media only screen and (max-width: 1024px) {
  .site-navigation {
    gap: 0;
  }
}

/* Media Query for Tablets and Smaller Screens */
@media only screen and (max-width: 768px) {
  .site-actions {
    gap: 25px;
  }

  .sign-up,
  .sign-in {
    padding: 15px 20px;
    font-size: 14px;
    border-radius: 15px;
  }

  .site-navigation {
    grid-template-columns: 1fr 150px;
  }
}

/* Media Query for Mobile Devices */
@media only screen and (max-width: 480px) {
  .site-actions {
    gap: 10px;
  }

  .sign-up,
  .sign-in {
    padding: 5px 20px;
    font-size: 14px;
  }

  .logo {
    height: 100px;
  }

  .site-header-content {
    flex-direction: column;
  }

  .site-navigation {
    display: flex;
    margin-bottom: 15px;
    width: 100%;
  }
}
