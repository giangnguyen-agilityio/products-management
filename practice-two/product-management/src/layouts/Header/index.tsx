// Libraries
import { Container, Flex } from '@chakra-ui/react'

// Constants
import { navigationLinks } from '@constants'

// Components
import Logo from '@components/common/Logo'
import Navigation from '@components/common/Navigation'

// Assets
import logoWebsite from '@assets/images/logo_website.webp'

// Header Component
const Header = () => {
  return (
    <Container as="header" className="header" padding={4}>
      <Flex justifyContent="space-between">
        <Logo imageSrc={logoWebsite} />
        <Navigation links={navigationLinks} />
      </Flex>
    </Container>
  )
}

export default Header
