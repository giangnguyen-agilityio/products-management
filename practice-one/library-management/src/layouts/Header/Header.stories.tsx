import {type StoryFn, type Meta} from '@storybook/react'
import {BrowserRouter} from 'react-router-dom'
import Header from './index'
import React from 'react'

export default {
  title: 'Example/Header',
  component: Header,
} as Meta<typeof Header>

const Template: StoryFn<typeof Header> = args => (
  <BrowserRouter>
    <Header {...args} />
  </BrowserRouter>
)
export const Default = Template.bind({})
Default.args = {
  isLogin: false,
}

export const Secondary = Template.bind({})
Secondary.args = {
  isLogin: true,
}
