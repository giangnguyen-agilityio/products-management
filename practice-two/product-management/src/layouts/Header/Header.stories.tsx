import type { Meta, StoryObj } from '@storybook/react'
import Header from './index'

const meta = {
  title: 'Practice Two/Header Component',
  component: Header,
  decorators: [
    Story => (
      <div
        style={{
          backgroundColor: '#532EE7',
          color: '#FFF',
        }}
      >
        <Story />
      </div>
    ),
  ],
  parameters: {
    layout: 'top',
  },
  tags: ['autodocs'],
} satisfies Meta<typeof Header>

export default meta
type Story = StoryObj<typeof meta>

export const Primary: Story = {}
