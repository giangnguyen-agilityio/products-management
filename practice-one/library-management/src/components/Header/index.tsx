import './header.css'
import Logo from '../Logo/index'
import logo from '../../assets/images/logo-website.png'
import Navigation from '../Navigation/index'
import UserProfile from '../UserProfile/index'
import { BrowserRouter as Router } from 'react-router-dom'
import Button from '../Button/index'

const Header = () => {
    const links = [
        { id: 'book', label: 'book', url: '/book' },
        { id: 'member', label: 'member', url: '/member' },
        { id: 'hire request', label: 'hire request', url: '/request' }
    ]

    const isLogin = false

    return (
        <header className="site-header">
            <div className="site-header-content">
                {/* The website logo */}
                <Logo imageSrc={logo} altText={'This is the logo website'} size={150} />
                {/* The website navigation */}
                <nav className="site-navigation">
                    <Router>
                        <Navigation links={links} />
                    </Router>
                    {/* The website actions */}
                    <div className="site-actions-wrapper">
                        {isLogin
                            ? (
                                <UserProfile avatarUrl={logo} email={'useremail@gmail.com'} />
                            )
                            : (
                                <div className="site-actions">
                                    <Button
                                        size="large"
                                        variant="secondary"
                                        className="sign-up"
                                        ariaLabel="Sign up"
                                    >
                                        SIGN UP
                                    </Button>
                                    <Button
                                        size="large"
                                        variant="primary"
                                        className="sign-in"
                                        ariaLabel="Sign in"
                                    >
                                        SIGN IN
                                    </Button>
                                </div>
                            )}
                    </div>
                </nav>
            </div>
        </header>
    )
}

export default Header
