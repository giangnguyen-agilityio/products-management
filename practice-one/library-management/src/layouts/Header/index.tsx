import React from 'react'
import logo from '@assets/images/logo-website.png'
import Logo from '@components/commons/Logo'
import Navigation from '@components/Navigation'
import UserProfile from '@components/UserProfile'
import Button from '@components/commons/Button'
import {linksForAdmin, linksForUser} from '@constants/navigationLinks'
import {getItemInLocalStorage} from '@helpers'
import {ROLE} from '@constants'
import './header.css'

interface HeaderProps {
  isLogin: boolean
  onSignIn: () => void
  onLogout: () => void
}

const Header: React.FC<HeaderProps> = props => {
  const {isLogin, onSignIn, onLogout} = props
  const isAdmin: boolean = getItemInLocalStorage('memberRole') === ROLE.ADMIN

  return (
    <header className="site-header">
      <div className="site-header-content">
        {/* The website logo */}
        <Logo
          imageSrc={logo}
          altText="This is the logo website"
          widthSize={150}
          heightSize={150}
        />
        {/* The website navigation */}
        <nav className="site-navigation">
          {/* Render navigation links */}
          <Navigation links={isAdmin ? linksForAdmin : linksForUser} />
          {/* The website actions */}
          <div className="site-actions-wrapper">
            {/* Render user profile if logged in */}
            {isLogin ? (
              <UserProfile
                avatarUrl={logo}
                email="useremail@gmail.com"
                onLogout={onLogout}
              />
            ) : (
              <div className="site-actions">
                {/* Render sign up button */}
                <Button
                  size="large"
                  variant="secondary"
                  className="sign-up"
                  ariaLabel="Sign up"
                >
                  SIGN UP
                </Button>
                {/* Render sign in button */}
                <Button
                  size="large"
                  variant="primary"
                  className="sign-in"
                  ariaLabel="Sign in"
                  onClick={onSignIn}
                >
                  SIGN IN
                </Button>
              </div>
            )}
          </div>
        </nav>
      </div>
    </header>
  )
}

export default Header
