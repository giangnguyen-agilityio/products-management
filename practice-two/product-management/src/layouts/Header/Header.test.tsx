import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'
import Header from '.'

describe('Header component', () => {
  it('renders without crashing', () => {
    render(
      <ChakraProvider>
        <Header />
      </ChakraProvider>
    )
  })

  it('renders logo', () => {
    const { getByAltText } = render(
      <ChakraProvider>
        <Header />
      </ChakraProvider>
    )

    const logo = getByAltText(/Website Logo/i)
    expect(logo).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { container } = render(
      <ChakraProvider>
        <Header />
      </ChakraProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
