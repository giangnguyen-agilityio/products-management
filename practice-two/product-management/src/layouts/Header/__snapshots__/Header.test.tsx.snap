// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div>
  <header
    class="chakra-container header css-1mr40xt"
  >
    <div
      class="css-gg4vpm"
    >
      <a
        href="/"
      >
        <div
          class="logo css-759u60"
        />
      </a>
      <button
        aria-label="Open Menu"
        class="chakra-button hamburger-menu css-1lguxu9"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </header>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
